File: src/main/java/config/TestStackProperties.java
package config;

import static config.TestConfig.CONFIG;

public class TestStackProperties {
    private static final String ENVIRONMENT = "environment";
    private static final String FRAMEWORK = "framework";
    private static final String WEB_URL = "web_url";
    private static final String API_URL = "api_url";
    private static final String ONEAPP_ANDROID_PATH = "oneapp_android_path";
    private static final String ONEAPP_IOS_PATH = "oneapp_ios_path";

    public static String getEnvironment() {
        return CONFIG.getProperty(ENVIRONMENT);
    }

    public static String getFramework() {
        return CONFIG.getProperty(FRAMEWORK);
    }

    public static String getWebUrl() {
        return CONFIG.getProperty(WEB_URL);
    }

    public static String getApiUrl() {
        return CONFIG.getProperty(API_URL);
    }

    public static String getOneappAndroidPath() {
        return CONFIG.getProperty(ONEAPP_ANDROID_PATH);
    }

    public static String getOneappIosPath() {
        return CONFIG.getProperty(ONEAPP_IOS_PATH);
    }
}

File: ts-playwright-test-automation-framework/src/main/code/config/TestStackProperties.ts
import { TestConfig } from "./TestConfig";

export class TestStackProperties {
    private static readonly ENVIRONMENT: string = "environment";
    private static readonly PLATFORM: string = "platform";
    private static readonly WEB_URL: string = "web_url";
    private static readonly API_URL: string = "api_url";
    private static readonly REMOTE_BROWSER_WS_URL: string = "remote_browser_ws_url";

    public static getEnvironment(): string {
        return TestConfig.CONFIG.getProperty(this.ENVIRONMENT)!;
    }

    public static getPlatform(): string {
        return TestConfig.CONFIG.getProperty(this.PLATFORM)!;
    }

    public static getWebUrl(): string {
        return TestConfig.CONFIG.getProperty(this.WEB_URL)!;
    }

    public static getApiUrl(): string {
        return TestConfig.CONFIG.getProperty(this.API_URL)!;
    }

    public static getRemoteBrowserWsUrl(): string {
        return TestConfig.CONFIG.getProperty(this.REMOTE_BROWSER_WS_URL)!;
    }
}